feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/common/NovaCardRestrictionCheckMixin.kt

```diff
@@ -0,0 +1,73 @@
+package io.novafoundation.nova.feature_assets.presentation.novacard.common
+
+import io.novafoundation.nova.common.mixin.restrictions.RestrictionCheckMixin
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.view.bottomSheet.action.ActionBottomSheetLauncher
+import io.novafoundation.nova.common.view.bottomSheet.action.ButtonPreferences
+import io.novafoundation.nova.common.view.bottomSheet.action.primary
+import io.novafoundation.nova.feature_account_api.domain.interfaces.SelectedAccountUseCase
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MultisigMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.ProxiedMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.hasChainAccountIn
+import io.novafoundation.nova.feature_account_api.domain.model.isThreshold1
+import io.novafoundation.nova.feature_assets.R
+import io.novafoundation.nova.runtime.ext.ChainGeneses
+
+private const val NOVA_CARD_AVAILABLE_CHAIN_ID = ChainGeneses.POLKADOT
+
+class NovaCardRestrictionCheckMixin(
+    private val accountUseCase: SelectedAccountUseCase,
+    private val resourceManager: ResourceManager,
+    private val actionLauncher: ActionBottomSheetLauncher,
+) : RestrictionCheckMixin {
+
+    override suspend fun isRestricted(): Boolean {
+        val selectedAccount = accountUseCase.getSelectedMetaAccount()
+        return isRestrictedByMultisig(selectedAccount) || isRestrictedByProxied(selectedAccount)
+    }
+
+    override suspend fun checkRestrictionAndDo(action: () -> Unit) {
+        val selectedAccount = accountUseCase.getSelectedMetaAccount()
+
+        when {
+            isRestrictedByMultisig(selectedAccount) -> showMultisigWarning()
+            isRestrictedByProxied(selectedAccount) -> showProxiedWarning()
+            else -> action()
+        }
+    }
+
+    private fun isRestrictedByMultisig(metaAccount: MetaAccount): Boolean {
+        if (metaAccount !is MultisigMetaAccount) return false
+
+        val isThresholdNot1 = !metaAccount.isThreshold1()
+        val chainNotAvailable = !metaAccount.hasChainAccountIn(NOVA_CARD_AVAILABLE_CHAIN_ID)
```

Comment:
We do not want to check whether meta account has chain accout, but want to check whether there is an account in such chain in general, so we should use `hasAccountIn`
Using `hasChainAccount` implies the knowledge of internal logic behind `MultisigMetaAccount`, in particular that threhsold 1 multisig accounts have one **chain** account. This knowledge should not be exposed